client/src/app/gui.rs

```diff
@@ -95,14 +97,27 @@ impl Gui {
         self.build_top_bar(ctx);
         self.build_bottom_bar(ctx);
 
+        let should_refresh_plot = self.should_refresh_plot();
         match self.tab {
             Tab::Packets => self.packets_table.ui(ctx),
             Tab::RtpPackets => self.rtp_packets_table.ui(ctx),
             Tab::RtpStreamsTable => self.rtp_streams_table.ui(ctx),
-            Tab::RtpStreamsPlot => self.rtp_streams_plot.ui(ctx),
+            Tab::RtpStreamsPlot => self.rtp_streams_plot.ui(ctx, should_refresh_plot),
         };
     }
 
+    fn should_refresh_plot(&mut self) -> bool {
+        // For large amount data plot can be laggy, so we don't refresh on every ui update
+        if self.update_counter == u32::MAX {
+            self.update_counter = 0
+        }
+
+        let should_refresh = self.update_counter % 25 == 0;
+
+        self.update_counter += 1;
+        should_refresh
+    }
```

Comment:
I don't like that, especially considering that (I'm pretty sure) egui doesn't actually run these on every frame.

What about something like this: in `rtp_streams_plot.rs` you check if number of packets in stream changed from previous frame, and if it did, you refresh the points. Then there's not need for this frame counting.

You can test it, there probably will be a bit of a lag at the beginning when a lot of new packets is coming, but in general I hope it's gonna be alright. If not, for now we can stick with frame counting.

Reply:
love the idea